util/conditions/unstructured.go

```diff
@@ -47,3 +55,11 @@ func (c *unstructuredWrapper) GetConditions() clusterv1.Conditions {
 	}
 	return conditions
 }
+
+// SetConditions set the conditions into an Unstructured object.
+func (c *unstructuredWrapper) SetConditions(conditions clusterv1.Conditions) {
+	valueBytes, _ := json.Marshal(conditions)
+	v := make([]interface{}, len(conditions))
+	_ = json.Unmarshal(valueBytes, &v)
+	_ = unstructured.SetNestedField(c.Unstructured.Object, v, "status", "conditions")
+}
```

Comment:
Should we make `SetConditions` return an error?

Reply:
The Setter interface was intentionally designed without error so the usage in the controller is one simple line vs 3/4.
Now, the only case where Set could lead to an error is unstructured setter, and mostly due to how set NestedFields works internally. I really would like to avoid returning an error for this specific use case...

Reply:
Can we at least log an error?

Reply:
I think we can use reflection to convert types as required by SetNestedField

Reply:
We can use https://pkg.go.dev/github.com/mitchellh/mapstructure?tab=doc

Reply:
fixed using runtime.DefaultUnstructuredConverter.ToUnstructured